Package.swift

```diff
@@ -0,0 +1,51 @@
+// swift-tools-version:5.3
+
+import PackageDescription
+
+let package = Package(
+    name: "OpenCV",
+    platforms: [
+        .macOS(.v10_12), .iOS(.v9),
+    ],
+    products: [
+        .library(
+            name: "OpenCV",
+            targets: ["OpenCV"]),
+    ],
+    dependencies: [],
+    targets: [
+        .target(
+            name: "OpenCV",
+            dependencies: ["opencv2"],
+            path: "modules/core/misc/objc/swift-package-manager/Sources",
+            linkerSettings: [
+                .linkedLibrary("c++"),
+                .linkedFramework("Accelerate"),
+                .linkedFramework("OpenCL", .when(platforms: [.macOS]))
+            ]
+        ),
+        // Recompute checksum via `swift package --package-path /path/to/opencv compute-checksum /path/to/opencv2.xcframework.zip`
+        .binaryTarget(
+            name: "opencv2",
+            url: "https://github.com/Rightpoint/opencv/releases/download/4.5.1/opencv2-4.5.1-dynamic.xcframework.zip",
+            checksum: "48273710fe03eb6d6a77ca57f96ef7917db395e6c3bc62e2b495df3dc8f1a0a9"
+        ),
+        // If you are compiling OpenCV locally, you can uncomment the below block to use a custom copy
+        // e.g. `$ python platforms/apple/build_xcframework.py --dynamic build/dynamic-xcframework`
+//        .binaryTarget(
+//            name: "opencv2",
+//            path: "build/dynamic-xcframework/opencv2.xcframework"
+//        ),
+        .testTarget(
+            name: "OpenCVTests",
+            dependencies: ["OpenCV"],
+            path: "modules/core/misc/objc/test/swift",
```

Comment:
There are several tests besides of "core" module which are not mentioned here.
Full set of tests is prepared during the build.

If we are moving to dedicated repository for the package it make sense to adapt build scripts to prepare additional folder with:
- `Package.swift` file based on template from `platforms/apple/Package.swift.template` (or `.in` suffix) with correct checksum and adjusted tests path
- copy of test files from build directory
- misc files, like readme.md (copy content from `platforms/apple/opencv-swift`)

This folder should be uploaded as artifact (`.zip` archive should be OK).
This artifact can be used later to prepare an update (commit/PR) in dedicated "opencv-swift" repository.

---

Note1: All names/paths above are just examples for now.
Note2: Prefer to keep code in Python scripts instead of workflows (it is very hard to debug "workflows" or other CI's scripts)

Reply:
Ahh I wasn't aware of how the OpenCV test process works, I'll take a closer look. I'm going to break this PR into smaller functional components, because it seems like the right path forward.

* GitHub Actions for building xcframework artifacts
* Fix Size/Point/Rect symbol conflict issue
* Package.swift in separate repository